identity/migrations/00000000000000_diesel_initial_setup/up.sql

```diff
@@ -0,0 +1,62 @@
+-- This file was automatically created by Diesel to setup helper functions
+-- and other internal bookkeeping. This file is safe to edit, any future
+-- changes will be added to existing projects as new migrations.
+
+
+
+
+-- Sets up a trigger for the given table to automatically set a column called
+-- `updated_at` whenever the row is modified (unless `updated_at` was included
+-- in the modified columns)
+--
+-- # Example
+--
+-- ```sql
+-- CREATE TABLE users (id SERIAL PRIMARY KEY, updated_at TIMESTAMP NOT NULL DEFAULT NOW());
+--
+-- SELECT diesel_manage_updated_at('users');
+-- ```
+CREATE OR REPLACE FUNCTION diesel_manage_updated_at(_tbl regclass) RETURNS VOID AS $$
+BEGIN
+    EXECUTE format('CREATE TRIGGER set_updated_at BEFORE UPDATE ON %s
+                    FOR EACH ROW EXECUTE PROCEDURE diesel_set_updated_at()', _tbl);
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION diesel_set_updated_at() RETURNS trigger AS $$
+BEGIN
+    IF (
+        NEW IS DISTINCT FROM OLD AND
+        NEW.updated_at IS NOT DISTINCT FROM OLD.updated_at
+    ) THEN
+        NEW.updated_at := current_timestamp;
+    END IF;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TABLE users (
```

Comment:
What benefit do you see from splitting them up into different steps? The current service implementation relies on them. If you split them up in say three different migrations, then a revert would break the service.

Reply:
1. I understood having the separate migration steps as some kind of "separation of concerns". So in the step `0000..00_diesel_initial_setup` you actually only do the initial diesel setup. And then in the 2nd step you set up your  three custom tables.
2. If you revert the one migration step you have now it will also fail, right?

Reply:
Number 1 is correct when you just start building your project, but I already start with a working service. Yes, to number 2. Anyway, I moved the initialization to an explicit migration.

Reply:
Don't be that easy on me...
I need some pushback :P